dts/bindings/pinctrl/ambiq,apollo4-pinctrl.yaml

```diff
@@ -0,0 +1,107 @@
+
+    The node has the 'pinctrl' node label set in your SoC's devicetree,
+    so you can modify it like this:
+
+      &pinctrl {
+              /* your modifications go here */
+      };
+
+    All device pin configurations should be placed in child nodes of the
+    'pinctrl' node, as shown in this example:
+
+      /* You can put this in places like a board-pinctrl.dtsi file in
+       * your board directory, or a devicetree overlay in your application.
+       */
+
+      /* include pre-defined combinations for the SoC variant used by the board */
+      #include <dt-bindings/pinctrl/ambiq-apollo4-pinctrl.h>
+
+      &pinctrl {
+        uart0_default: uart0_default {
+          group1 {
+            pinmux = <UART0TX_P60>;
+          };
+          group2 {
+            pinmux = <UART0RX_P47>;
+            input-enable;
+          };
+        };
+      };
+
+    The 'uart0_default' child node encodes the pin configurations for a
+    particular state of a device; in this case, the default (that is, active)
+    state.
+
+    As shown, pin configurations are organized in groups within each child node.
+    Each group can specify a list of pin function selections in the 'pinmux'
+    property.
+
+    A group can also specify shared pin properties common to all the specified
+    pins, such as the 'input-enable' property in group 2.
+
+compatible: "ambiq,apollo4-pinctrl"
+
+include: base.yaml
+
+child-binding:
+  description: |
+    Definitions for a pinctrl state.
+  child-binding:
+
+    include:
+      - name: pincfg-node.yaml
+        property-allowlist:
+          - input-enable
+          - drive-push-pull
+          - drive-open-drain
+          - bias-high-impedance
+
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of pins sharing the same group properties. Each
+          element of the array is an integer constructed from the
+          pin number and the alternative function of the pin.
+      drive-strength:
+        type: string
+        enum:
+          - "0.1"
+          - "0.5"
+          - "0.75"
+          - "1.0"
+        default: "0.1"
+        description: |
+          The drive strength of a pin, relative to full-driver strength.
+          The default value is 0.1, which is the reset value.
+      slew-rate:
+        type: string
+        enum:
+          - "slow"
+          - "fast"
+        default: "slow"
+        description: |
+          Select slew rate for a pin. The default is slow, which is the reset value.
+      bias-value:
+        type: string
+        enum:
+          - "0"
+          - "pulldown_50k"
+          - "pullup_1_5k"
+          - "pullup_6k"
+          - "pullup_12k"
+          - "pullup_12k"
+          - "pullup_24k"
+          - "pullup_50k"
+          - "pullup_100k"
+        default: "0"
+        description: |
+          The pullup/pulldown resistor value. The default value is reset value.
```

Comment:
pull-up/down enablement should still follow `bias-pull-up/down`. Value may be selected using an extra property, e.g. `ambiq,pull-up-ohms` and value be in ohms (you can still use an enumerated integer to rely on DT enum macros if preferred)

Reply:
Added `bias-pull-up/down`. Also added `ambiq,pull-up-ohms` for pull-up resitor value configuration.